src/server/main_service.cc

```diff
@@ -590,6 +600,42 @@ void Service::Init(util::AcceptServer* acceptor, std::vector<facade::Listener*>
   GenericFamily::Init(&pp_);
   server_family_.Init(acceptor, std::move(listeners), &cluster_family_);
 
+  if (string flag = GetFlag(FLAGS_replicaof); !flag.empty()) {
+    string ip, port;
+    {
+      auto pos = flag.find_last_of(':');
+      CHECK(pos != string::npos) << "Invalid format for --replicaof: missing ':'.";
+
+      string ip1 = flag.substr(0, pos);
+      port = flag.substr(pos + 1);
+
+      CHECK(!ip1.empty() && !port.empty())
+          << "Invalid format for --replicaof: IP or port are empty.";
+
+      // For IPv6
+      CHECK(!((ip1.front() == '[') ^ (ip1.back() == ']')))
+          << "Invalid format for --replicaof: unclosed brackets.";
+
+      if (ip1.front() == '[') {
+        ip = ip1.substr(1, ip1.length() - 2);
+        LOG(INFO) << "Received IPv6: " << ip;
+      } else {
+        ip = move(ip1);
+        LOG(INFO) << "Received IPv4: " << ip;
+      }
+    }
+
+    LOG(INFO) << "Replicating instance at " << ip << ":" << port;
+
+    bool success = false;
+    pp_[0].Await([this, ip = move(ip), port = move(port), &success]() {
```

Comment:
Since you're `Await()`ing here, you could capture `ip` and `port` by reference instead of `move()`ing them inside

Reply:
Is there any reason to leave resources alive and capture by reference, rather than letting them expire after the lambda ends?

Reply:
Capturing by reference means that the lifetime of the variables doesn't change (they live regardless of the lambda), so they are unaffected.
However, on the lambda-side, captured variables are by-reference, meaning they are not copied internally but instead they only save a pointer to the real variable outside the lambda.